src/shared/auth.ts

```diff
@@ -14,77 +15,75 @@ export const SafeUrlSchema = z.string().url()
 
       return z.NEVER;
     }
-  }).refine(
+  })
+  .refine(
     (url) => {
       const u = new URL(url);
-      return u.protocol !== 'javascript:' && u.protocol !== 'data:' && u.protocol !== 'vbscript:';
+      return (
+        u.protocol !== "javascript:" &&
+        u.protocol !== "data:" &&
+        u.protocol !== "vbscript:"
+      );
     },
     { message: "URL cannot use javascript:, data:, or vbscript: scheme" }
-);
-
+  );
 
 /**
  * RFC 9728 OAuth Protected Resource Metadata
  */
-export const OAuthProtectedResourceMetadataSchema = z
-  .object({
-    resource: z.string().url(),
-    authorization_servers: z.array(SafeUrlSchema).optional(),
-    jwks_uri: z.string().url().optional(),
-    scopes_supported: z.array(z.string()).optional(),
-    bearer_methods_supported: z.array(z.string()).optional(),
-    resource_signing_alg_values_supported: z.array(z.string()).optional(),
-    resource_name: z.string().optional(),
-    resource_documentation: z.string().optional(),
-    resource_policy_uri: z.string().url().optional(),
-    resource_tos_uri: z.string().url().optional(),
-    tls_client_certificate_bound_access_tokens: z.boolean().optional(),
-    authorization_details_types_supported: z.array(z.string()).optional(),
-    dpop_signing_alg_values_supported: z.array(z.string()).optional(),
-    dpop_bound_access_tokens_required: z.boolean().optional(),
-  })
-  .passthrough();
+export const OAuthProtectedResourceMetadataSchema = z.object({
```

Comment:
I think `looseObject` replaces the removed `passthrough`
```suggestion
export const OAuthProtectedResourceMetadataSchema = z.looseObject({
```

Reply:
Yes, good catch: https://zod.dev/v4/changelog?id=deprecates-strict-and-passthrough